lib/lfrfid/lfrfid_raw_file.c

```diff
@@ -0,0 +1,149 @@
+#include "lfrfid_raw_file.h"
+#include "tools/varint_pair.h"
+#include <toolbox/stream/file_stream.h>
+#include <toolbox/varint.h>
+
+#define LFRFID_RAW_FILE_MAGIC 0x4C464952
+#define LFRFID_RAW_FILE_VERSION 1
+
+#define TAG "RFID RAW File"
+
+typedef struct {
+    uint32_t magic;
+    uint32_t version;
+    float frequency;
+    float duty_cycle;
+    uint32_t max_buffer_size;
+} LFRFIDRawFileHeader;
+
+struct LFRFIDRawFile {
+    Stream* stream;
+    uint32_t max_buffer_size;
+
+    uint8_t* buffer;
+    uint32_t buffer_size;
+    size_t buffer_counter;
+};
+
+LFRFIDRawFile* lfrfid_raw_file_alloc(Storage* storage) {
+    LFRFIDRawFile* file = malloc(sizeof(LFRFIDRawFile));
+    file->stream = file_stream_alloc(storage);
+    file->buffer = NULL;
+    return file;
+}
+
+void lfrfid_raw_file_free(LFRFIDRawFile* file) {
+    if(file->buffer) free(file->buffer);
+    stream_free(file->stream);
+    free(file);
+}
+
+bool lfrfid_raw_file_open_write(LFRFIDRawFile* file, const char* file_path) {
+    return file_stream_open(file->stream, file_path, FSAM_READ_WRITE, FSOM_CREATE_ALWAYS);
+}
+
+bool lfrfid_raw_file_open_read(LFRFIDRawFile* file, const char* file_path) {
+    return file_stream_open(file->stream, file_path, FSAM_READ, FSOM_OPEN_EXISTING);
+}
+
+bool lfrfid_raw_file_write_header(
+    LFRFIDRawFile* file,
+    float frequency,
+    float duty_cycle,
+    uint32_t max_buffer_size) {
+    LFRFIDRawFileHeader header = {
+        .magic = LFRFID_RAW_FILE_MAGIC,
+        .version = LFRFID_RAW_FILE_VERSION,
+        .frequency = frequency,
+        .duty_cycle = duty_cycle,
+        .max_buffer_size = max_buffer_size};
+
+    size_t size = stream_write(file->stream, (uint8_t*)&header, sizeof(LFRFIDRawFileHeader));
+    if(size == sizeof(LFRFIDRawFileHeader)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+bool lfrfid_raw_file_write_buffer(LFRFIDRawFile* file, uint8_t* buffer_data, size_t buffer_size) {
+    size_t size;
+    size = stream_write(file->stream, (uint8_t*)&buffer_size, sizeof(size_t));
+    if(size != sizeof(size_t)) return false;
+
+    size = stream_write(file->stream, buffer_data, buffer_size);
+    if(size != buffer_size) return false;
+
+    return true;
```

Comment:
return (stream_write() == sizeof()) && (stream_write() == buffer_size)
would be easier to read. Personal opinion, completely optional